python/main.py

```diff
@@ -23,9 +25,51 @@ def root():
     return {"message": "Hello, world!"}
 
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+async def add_item(name: str = Form(...), category: str = Form(...), image: UploadFile = File(...)):
+    logger.info(f"Receive item, name: {name}, category: {category}")
+
+    # hash image
+    imageContent = await image.read()
+    hashedImage = hashlib.sha256(imageContent).hexdigest()
+    hashedImageName = hashedImage + os.path.splitext(image.filename)[1]
+
+    # update items.json file
+    newItem = {
+        'name': name,
+        'category': category,
+        'image_filename': hashedImageName
+    }
+
+    with open("items.json", 'r') as itemsFile:
```

Comment:
Let's introduce a constant `ItemsFileName` (the naming is up to you) so you don't have to type `items.json` every time! It will reduce human errors and improves scalability!